components/sup/protocols/event.proto

```diff
@@ -0,0 +1,74 @@
+syntax = "proto3";
+
+package chef.habitat.supervisor.event;
+import "google/protobuf/timestamp.proto";
+import "google/protobuf/duration.proto";
+
+message EventMetadata {
+  // The unique ID of this Supervisor.
+  string supervisor_id = 1;
+  // The IP address on which the Supervisor is listening for gossip.
+  string ip_address = 2;
+  // The time at which this event occurred.
+  google.protobuf.Timestamp timestamp = 3;
+  string application = 4;
+  string environment = 5;
+  map<string, string> meta = 6;
+}
+
+message ServiceMetadata {
+  // todo: make a FQPI type
+  string package_ident = 1;
+  // not necessarily FQPI
+  string spec_ident = 2;
+  // todo: make / reuse a real type?
+  string service_group = 3;
+  // The channel a service was initially downloaded from, and (if an
+  // update strategy is set) the channel from which updates will be
+  // pulled.
+  string update_channel = 4;
```

Comment:
This is really just documenting existing behavior, describing the `--channel` option behavior in `hab svc load <MY_PACKAGE> --channel=foo`. With _that_ invocation, you're going to grab the latest version from the `foo` channel. If you were to also add an `--strategy`, then it would _also_ be the channel we'd check for additional updates.

I could try modifying this to only set an `update_channel` value if the `update_strategy` is something other than "none". That would be more direct, at least.

(related: I'm somewhat bummed that `UpdateStrategy::None` exists, rather than `Option<UpdateStrategy>` being used 😦 )

Reply:
Thinking a bit further, I'm tempted to create a new composite type that has a non-None update strategy _and_ a channel, and use _that_ instead, since you really want to have both.

Initially, it would only exist at the Protobuf level, but I think a good argument could be made for extending such a construct into the rest of the Supervisor at some point, which could clarify and simplify a lot of other stuff.

Reply:
I'm not sure how to feel about this sort of overloading of the `--channel` arg. If you do do a `hab svc load --channel foo bar/baz` and `bar/baz` is already on disk, but it didn't come from channel `foo`, you can end up with a different version than if you had run the same command *without* having the package already installed.

I understand this is more of an issue with our existing implementation. Maybe the docs here should just say that it's the value of the `--channel` arg supplied to `hab svc load`, which may or may not be the same as the `--channel` arg which may or may not have been supplied to `hab pkg install`.

Reply:
I think the current behavior in the situation you describe is that hab would end up getting a more-recent version from the `foo` channel, or that it would say what's on disk is more recent than what's in `foo` and use that instead.

The overloading is definitely a weird situation. In this particular use case, though, the desire is to know which services stand to be affected by promotions in a given channel, so it really does deal with updates exclusively. I'm going to prototype a new type that expresses that and see what it ends up looking like.

Reply:
@baumanj how does this reformulation strike you? https://github.com/habitat-sh/habitat/pull/6344/commits/d34100083b9efc9e01446d5f9a9714a2a9ac6320